File: LanguAI.Backend/LanguAI.Backend/ViewModels/User/UserDataViewModel.cs
using LanguAI.Backend.ViewModels.Learning;

namespace LanguAI.Backend.ViewModels.User;

public class UserDataViewModel
{
    public int Id { get; set; }

    public string Username { get; set; }

    public DateTime DateOfBirth { get; set; }

    public int LanguageId { get; set; }

    public int Streak { get; set; }

    public LearningViewModel CurrentLearning { get; set; }
}

File: LanguAI.Frontend/src/api/models/user-data-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { LearningViewModel } from '../models/learning-view-model';
export interface UserDataViewModel {
  currentLearning?: LearningViewModel;
  dateOfBirth?: string;
  id?: number;
  languageId?: number;
  streak?: number;
  username?: string | null;
}
